src/main/java/gregtech/integration/jei/recipe/primitive/MaterialTree.java

```diff
@@ -0,0 +1,123 @@
+package gregtech.integration.jei.recipe.primitive;
+
+import com.google.common.collect.ImmutableList;
+import gregtech.api.unification.OreDictUnifier;
+import gregtech.api.unification.material.type.DustMaterial;
+import gregtech.api.unification.material.type.IngotMaterial;
+import gregtech.api.unification.ore.OrePrefix;
+import mezz.jei.api.ingredients.IIngredients;
+import mezz.jei.api.recipe.IRecipeWrapper;
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.fluids.FluidStack;
+
+import java.util.List;
+import java.util.ArrayList;
+
+public class MaterialTree implements IRecipeWrapper {
```

Comment:
One thing that I think would be good for this class is a tooltip method for adding tooltips to the items in the material page to notify users of how the materials are made. For example an Aluminum Rod would have the tooltip: 

```
Made from Aluminum ingot in: Extruder, Lathe
Made from Aluminum Long Rod in Cutting Saw
```

In addition, tooltips would give you the opportunity to mention any recipe properties. You have a line showing the EBF coils, however for the fluid, if it was made in a Fusion Reactor, you could have a tooltip on it:

`Made in Fusion Reactor MK 2`